cli/azd/main.go

```diff
@@ -34,6 +35,8 @@ func main() {
 		log.SetOutput(io.Discard)
 	}
 
+	container.RegisterDependencies()
```

Comment:
Did you consider simply creating a package called `providers` under `provisioning` with a simple `NewProvider` factory that allows creating all known provider types? `providers` can consume all of the known concretion types.

I think the current change is structured more as a plugin-model, which is fine but introduces more indirection. Just a suggestion, not a huge deal either way.

Reply:
This still creates a circular dependency between `provisioning` and `providers` package since the provisioning manager needs to leverage the factory internally based on the configuration options.